packages/mocks/src/getMockData.ts

```diff
@@ -12,24 +12,25 @@ const mockData: Record<string, object> = {
 /**
  * Get a mock response for a given request and path parameters.
  *
+ * @template T - The expected type of the response body (defaults to `object`)
  * @param {Object} param0 - The object containing the request parameters
  * @param {string} param0.pathname - The incoming request pathname to match against the mock handlers
- * @param {string} param0.method - An optional string, the incoming request method to match against the mock handlers
- * @param {PathParams} param0.params - An optional object containing the path parameters of the request
- * @returns {MockResponse | null} A mock response or null if no mock handler is found
+ * @param {string} [param0.method] - An optional string, the incoming request method to match against the mock handlers
+ * @param {PathParams} [param0.params] - An optional object containing the path parameters of the request
+ * @returns {MockResponse<T> | null} A mock response or null if no mock handler is found
  */
-export const getMockData = ({ pathname }: MockDataRequestParameters): MockResponse | null => {
+export const getMockData = <T = object>({
+  pathname,
+}: MockDataRequestParameters): MockResponse<T> | null => {
+  // Dynamic handlers, using custom logic.
+
   // Add your mock handlers here, e.g.:
   // if (method === 'GET' && pathname === '/api/some-resource' && params['details'] === '1') {
   //   return mockResponse;
   // }
 
-  // Dynamic handlers, using custom logic.
-
-  // ...
-
   // Static handlers, using mockData.
-  const data = mockData[pathname] || null;
+  const data = (mockData[pathname] as T) || null;
```

Comment:
@yaacov since we are doing a cast here it only gives a false sense of type security

Reply:
hmm, do you think I should remove the generic, i like it ... do you think it's important not to have it ?

Reply:
the use case i see is in tests, where you know what the getMock method return to you ... and it should fail if it gets the wrong type ... I think it's ok

on the handler, it just return object, so no false type there

Reply:
basically those 2 lines below are doing the same:
```
const response = getMockData<ProvidersInventory>(requestParameters);
 const response = getMockData(requestParameters) as ProvidersInventory;

```
I like the second more because it explicitly shows what's happening: a forced type cast

Reply:
ok, I'll remove the generic :-(